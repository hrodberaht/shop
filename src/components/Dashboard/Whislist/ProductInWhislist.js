import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ProductInWhislist extends Component {
  handleClick = () => {
    const {
      remove,
      product: { productId },
    } = this.props;
    remove(productId);
  };

  render() {
    const {
      product: {
        name, price, pcsOrder, totalPrice,
      },
    } = this.props;
    return (
      <tr>
        <td>{name}</td>
        <td>{price}</td>
        <td>{pcsOrder}</td>
        <td>
          <span>$</span>
          {totalPrice}
        </td>
        <td>
          <button type="button" className="btn btn-danger" onClick={this.handleClick}>
            X
          </button>
        </td>
      </tr>
    );
  }
}

ProductInWhislist.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string,
    price: PropTypes.number,
  }).isRequired,
  remove: PropTypes.func.isRequired,
};
