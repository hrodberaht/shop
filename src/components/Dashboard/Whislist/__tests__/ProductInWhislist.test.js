import ProductInWhislist from '../ProductInWhislist';

const defaultProps = {
  product: {
    productId: 'product-1',
    type: 'Printer',
    name: 'ZX3',
    price: 400,
    inStock: '30',
  },
  remove: jest.fn(),
};

const setup = buildSetup(ProductInWhislist, defaultProps);

describe('ProductInWhislist />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call remove after click', () => {
    const { wrapper, props } = setup();
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().props.remove).toHaveBeenCalledWith(props.product.productId);
  });
});
