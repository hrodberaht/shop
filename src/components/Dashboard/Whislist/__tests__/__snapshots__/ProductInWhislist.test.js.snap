// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInWhislist /> should render without crash 1`] = `
<tr>
  <td>
    ZX3
  </td>
  <td>
    400
  </td>
  <td />
  <td>
    <span>
      $
    </span>
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </td>
</tr>
`;
